keylime-agent/src/config.rs

```diff
@@ -59,244 +59,228 @@ pub static DEFAULT_AGENT_DATA_PATH: &str = "agent_data.json";
 pub static DEFAULT_CONFIG: &str = "/etc/keylime/agent.conf";
 pub static DEFAULT_CONFIG_SYS: &str = "/usr/etc/keylime/agent.conf";
 
-impl Source for KeylimeConfig {
-    fn collect(&self) -> Result<Map<String, Value>, ConfigError> {
-        let agent: Map<String, Value> = Map::from([
-            ("version".to_string(), self.agent.version.to_string().into()),
-            ("uuid".to_string(), self.agent.uuid.to_string().into()),
-            ("ip".to_string(), self.agent.ip.to_string().into()),
-            ("port".to_string(), Value::from(self.agent.port)),
-            (
-                "contact_ip".to_string(),
-                if let Some(ref s) = self.agent.contact_ip {
-                    s.to_string().into()
-                } else {
-                    "".into()
-                },
-            ),
-            (
-                "contact_port".to_string(),
-                if let Some(ref s) = self.agent.contact_port {
-                    s.to_string().into()
-                } else {
-                    "".into()
-                },
-            ),
-            (
-                "registrar_ip".to_string(),
-                self.agent.registrar_ip.to_string().into(),
-            ),
-            (
-                "registrar_port".to_string(),
-                self.agent.registrar_port.into(),
-            ),
-            (
-                "enable_agent_mtls".to_string(),
-                self.agent.enable_agent_mtls.into(),
-            ),
-            (
-                "keylime_dir".to_string(),
-                self.agent.keylime_dir.to_string().into(),
-            ),
-            (
-                "server_key".to_string(),
-                if let Some(ref s) = self.agent.server_key {
-                    s.to_string().into()
-                } else {
-                    "".into()
-                },
-            ),
-            (
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) struct AgentConfig {
+    pub version: Option<String>,
```

Comment:
Now everything is optional, can this be a problem?

Reply:
It is not a problem because since the last big configuration change to TOML format and to use the `config` crate, we always base the configuration from the defaults. So basically there is no case when an option is not set.